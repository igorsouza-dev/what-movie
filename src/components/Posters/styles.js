import styled from 'styled-components';
import { device, posterSize, screenSize } from 'styles/variables';

export const Container = styled.div`
  display: ${(props) => props.type};
  ${(props) => {
    if (props.type === 'flex') {
      return `
        overflow-x: auto;
        min-height: max-content;
      `;
    }
    return `

      @media only screen and (max-width: 409px) {
        display: flex;
        flex-direction: row;
        justify-content: flex-start;
        max-width: 300px;
        overflow-x: auto;
      }
      @media only screen and (min-width: 410px) and (max-width: 584px) {
        display: grid;
        grid-gap: 5px;
        padding: 10px;
        grid-template-columns: repeat(2, ${posterSize}px);
        max-width: ${screenSize.mobileL};
      }
      @media only screen and (min-width: 585px) and (max-width: 784px) {
        display: grid;
        grid-gap: 5px;
        padding: 10px;
        grid-template-columns: repeat(3, ${posterSize}px);
        max-width: ${screenSize.tablet};
      }
      @media only screen and (min-width: 785px) and (max-width: 1024px)  {
        display: grid;
        grid-gap: 5px;
        padding: 10px;
        grid-template-columns: repeat(4, ${posterSize}px);
        max-width: ${screenSize.laptop};
      }
      @media only screen and (min-width: 1025px) {
        display: grid;
        grid-gap: 5px;
        padding: 10px;
        grid-template-columns: repeat(5, ${posterSize}px);
        max-width: ${screenSize.laptop};
      }
      @media only screen and (min-width: 1440px) {
        display: grid;
        grid-gap: 5px;
        padding: 10px;
        grid-template-columns: repeat(6, ${posterSize}px);
        max-width: ${screenSize.laptop};
      }
    `;
  }}

  padding: 10px;
`;
