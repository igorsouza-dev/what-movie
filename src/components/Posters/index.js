import React from 'react';
import PropTypes from 'prop-types';
import Poster from 'components/Poster';
import { Container } from './styles';

function Posters({ movies, type, showButtons }) {
  return (
    <Container type={type}>
      {movies.map((movie) => (
        <Poster movie={movie} key={movie.id} showButtons={showButtons} />
      ))}
    </Container>
  );
}
Posters.propTypes = {
  type: PropTypes.string,
  movies: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
    }),
  ).isRequired,
};
Posters.defaultProps = {
  type: 'flex',
};
export default Posters;
