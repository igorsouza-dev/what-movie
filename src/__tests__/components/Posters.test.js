import React from 'react';
import { cleanup, render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import Posters from 'components/Posters';

import { renderWithRedux } from '../utils/redux';

describe('<Posters />', () => {
  afterEach(cleanup);
  const movies = [
    { id: 1, url: 'http://test.url.com/1', original_title: 'Test movie 1' },
    { id: 2, url: 'http://test.url.com/2', original_title: 'Test movie 2' },
    { id: 3, url: 'http://test.url.com/3', original_title: 'Test movie 3' },
  ];
  const history = createMemoryHistory();
  it('should render', () => {
    const { container, getByText } = render(
      <Router history={history}>
        <Posters movies={movies} />
      </Router>
    );
    expect(getByText('Test movie 1')).toBeInTheDocument();
    expect(getByText('Test movie 2')).toBeInTheDocument();
    expect(getByText('Test movie 3')).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
  it('should change display when passing prop', () => {
    const { container, getByText } = render(
      <Router history={history}>
        <Posters movies={movies} type="grid" />
      </Router>
    );
    expect(getByText('Test movie 1')).toBeInTheDocument();
    expect(getByText('Test movie 2')).toBeInTheDocument();
    expect(getByText('Test movie 3')).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
