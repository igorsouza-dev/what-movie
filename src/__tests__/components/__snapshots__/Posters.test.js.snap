// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posters /> should change display when passing prop 1`] = `
<div
  class="sc-jTzLTM dgMVdo"
  type="grid"
>
  <div
    class="sc-ifAKCX cRQFzp"
  >
    <div
      class="sc-bZQynM mjAAS"
    >
      /10
    </div>
    <div
      class="sc-VigVT iOQNnG"
      data-testid="poster-img-container"
    >
      <img
        alt="Test movie 1"
        src="http://test.url.com/1"
      />
    </div>
    <div
      class="sc-gzVnrw dFkHji"
    >
      <div
        class="sc-EHOje IcjOp"
        data-testid="name-container"
      >
        <strong>
          Test movie 1
        </strong>
      </div>
    </div>
  </div>
  <div
    class="sc-ifAKCX cRQFzp"
  >
    <div
      class="sc-bZQynM mjAAS"
    >
      /10
    </div>
    <div
      class="sc-VigVT iOQNnG"
      data-testid="poster-img-container"
    >
      <img
        alt="Test movie 2"
        src="http://test.url.com/2"
      />
    </div>
    <div
      class="sc-gzVnrw dFkHji"
    >
      <div
        class="sc-EHOje IcjOp"
        data-testid="name-container"
      >
        <strong>
          Test movie 2
        </strong>
      </div>
    </div>
  </div>
  <div
    class="sc-ifAKCX cRQFzp"
  >
    <div
      class="sc-bZQynM mjAAS"
    >
      /10
    </div>
    <div
      class="sc-VigVT iOQNnG"
      data-testid="poster-img-container"
    >
      <img
        alt="Test movie 3"
        src="http://test.url.com/3"
      />
    </div>
    <div
      class="sc-gzVnrw dFkHji"
    >
      <div
        class="sc-EHOje IcjOp"
        data-testid="name-container"
      >
        <strong>
          Test movie 3
        </strong>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Posters /> should render 1`] = `
<div
  class="sc-jTzLTM gjILoC"
  type="flex"
>
  <div
    class="sc-ifAKCX cRQFzp"
  >
    <div
      class="sc-bZQynM mjAAS"
    >
      /10
    </div>
    <div
      class="sc-VigVT iOQNnG"
      data-testid="poster-img-container"
    >
      <img
        alt="Test movie 1"
        src="http://test.url.com/1"
      />
    </div>
    <div
      class="sc-gzVnrw dFkHji"
    >
      <div
        class="sc-EHOje IcjOp"
        data-testid="name-container"
      >
        <strong>
          Test movie 1
        </strong>
      </div>
    </div>
  </div>
  <div
    class="sc-ifAKCX cRQFzp"
  >
    <div
      class="sc-bZQynM mjAAS"
    >
      /10
    </div>
    <div
      class="sc-VigVT iOQNnG"
      data-testid="poster-img-container"
    >
      <img
        alt="Test movie 2"
        src="http://test.url.com/2"
      />
    </div>
    <div
      class="sc-gzVnrw dFkHji"
    >
      <div
        class="sc-EHOje IcjOp"
        data-testid="name-container"
      >
        <strong>
          Test movie 2
        </strong>
      </div>
    </div>
  </div>
  <div
    class="sc-ifAKCX cRQFzp"
  >
    <div
      class="sc-bZQynM mjAAS"
    >
      /10
    </div>
    <div
      class="sc-VigVT iOQNnG"
      data-testid="poster-img-container"
    >
      <img
        alt="Test movie 3"
        src="http://test.url.com/3"
      />
    </div>
    <div
      class="sc-gzVnrw dFkHji"
    >
      <div
        class="sc-EHOje IcjOp"
        data-testid="name-container"
      >
        <strong>
          Test movie 3
        </strong>
      </div>
    </div>
  </div>
</div>
`;
